src/App.js

```diff
@@ -167,17 +183,38 @@ export default function App() {
             :)
           </p>
         </div>
-      ) : showCall ? (
+      );
+    }
+
+    // No API errors? Let's check our hair then.
+    if (showHairCheck) {
```

Comment:
can you make these else if's? _Just in case_ multiple ones were somehow truthy? (Actually I'm not sure if the error would show in addition to the other views, but I believe haircheck/call/home should be exclusive)

Reply:
HmmMmmm I prefer `if` statements for their readability over `if/else`. But I can see how claiming truthiness here is difficult since there are quite a few possible states:

- there is an API error (show error message)
- there is no API error, and the call has not been joined (HomeScreen or HairCheck)
- there is no API error, the call has not been joined, and `startHairCheck()` has not been called (HomeScreen)
- there is no API error, the call has not been joined, `startHairCheck()` has been called (HairCheck)
- there is no API error, `startHairCheck()` was called, and the call has been joined (Call)
I do believe all of the above scenarios are covered by these lines:

```
 const showCall =
    !apiError && [STATE_JOINING, STATE_JOINED, STATE_ERROR].includes(appState);
  const showHairCheck = !apiError && appState === STATE_HAIRCHECK;
```

(Where `STATE_ERROR` can actually be ignored, it's confusing and I should remove it or refactor it so that API errors fall under it)

Reply:
ah you know what I didn't even consider you're returning in each if statement anyway so it actually doesn't matter, ignore me!